languages/rust/oso/src/query.rs

```diff
@@ -18,15 +17,18 @@ impl Iterator for Query {
 pub struct Query {
     inner: polar_core::polar::Query,
     calls: HashMap<u64, PolarResultIter>,
-    host: Arc<Mutex<crate::host::Host>>,
+    host: Host,
+    /// Marker to make sure query is not safe to Send/Sync
+    _marker: std::ptr::NonNull<()>,
```

Comment:
This is undoing some of the unsafety caused by marking `Instance` as `Send + Sync`. 
Since `NonNull` is `!Send + !Sync`, this makes `Query` also those things.

The logic is: we're asserting (unsafely) our own invariants: `Instance` is safe to share/move between threads because it can only move between threads when on a `Host` instance on `Oso`. And the only way to add instances to `Oso`'s `Host` is when they are `Send + Sync`.

We could maybe make that more direct, by just asserting `unsafe impl Send for Oso`...